.github/workflows/validate-terraform.yml

```diff
@@ -0,0 +1,30 @@
+name: Validate Terraform
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  validate-terraform:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Terraform repo
+        uses: actions/checkout@v3
+
+      - name: Configure local backend
+        run: |
+          cp .github/workflows/backend_override.tf terraform/aws/
```

Comment:
Instead of this, you can pass `-backend-config` to `terraform init`

Reply:
I thought about this, but I'm not sure how to accomplish it here. 
The argument to `-backend-config` is a key-value vars file or "key=value". You can't pass a terraform configuration file, so the `terraform` block and `backend` setting must be part of the current configuration.  For the validation action, we change the backend type from `s3` to `local`.  I wasn't able to do that via `-backend-config` so I used the override.